kvstore/kvstore.go

```diff
@@ -0,0 +1,72 @@
+package kvstore
+
+import (
+	"errors"
+	"io"
+
+	"github.com/fastly/compute-sdk-go/internal/abi/fastly"
+)
+
+var ErrKeyNotFound = errors.New("kvstore: key not found")
+
+type Entry struct {
+	io.Reader
+
+	validString bool
+	s           string
+}
```

Comment:
Embedding an io.Reader in this way allows callers to treat Entry as an io.Reader directly, which doesn't seem to be correct, as methods like String consume from that Reader and mutate internal state in a way that callers aren't able to do.

This should probably be something like

```suggestion
type Entry struct {
	src  io.Reader
	read bool
	str  string
}
```

Reply:
Opened a follow-up issue on this. https://github.com/fastly/compute-sdk-go/issues/40